dotCMS/src/main/java/com/dotmarketing/common/reindex/ReindexThread.java

```diff
@@ -315,35 +360,44 @@ public static void unpause() {
         }
     }
 
-    private static void unpauseImpl() {
+    private static synchronized void unpauseImpl() {
+        ReindexThread instance = getInstance();
+        ThreadState currentState = instance.state.get();
 
-        ThreadState state = getInstance().state.get();
-        if (state == ThreadState.PAUSED) {
+        if (currentState == ThreadState.PAUSED) {
             Logger.info(ReindexThread.class, "--- Unpausing reindex thread ");
             cache.get().remove(REINDEX_THREAD_PAUSED);
-            getInstance().state(ThreadState.RUNNING);
-        } else if (state == ThreadState.STOPPED) {
+            instance.state.compareAndSet(ThreadState.PAUSED, ThreadState.RUNNING);
+        } else if (currentState == ThreadState.STOPPED) {
             Logger.info(ReindexThread.class, "--- Recreating ReindexThread from stopped");
             OSGISystem.getInstance().initializeFramework();
             Logger.infoEvery(ReindexThread.class, "--- ReindexThread Running", 60000);
             cache.get().remove(REINDEX_THREAD_PAUSED);
-            final Thread thread = new Thread(getInstance().ReindexThreadRunnable,
-                    "ReindexThreadRunnable");
 
             final DotSubmitter submitter = DotConcurrentFactory.getInstance()
-                    .getSubmitter("ReindexThreadSubmitter",
+                    .getSubmitter(
+                            "ReindexThreadSubmitter",
                             new DotConcurrentFactory.SubmitterConfigBuilder()
                                     .poolSize(1)
                                     .maxPoolSize(1)
                                     .queueCapacity(2)
-                                    .rejectedExecutionHandler(
-                                            new ThreadPoolExecutor.DiscardOldestPolicy())
-                                    .build()
-                    );
-            submitter.submit(thread);
-            getInstance().state(ThreadState.RUNNING);
-        }
+                                    .rejectedExecutionHandler(new ThreadPoolExecutor.DiscardOldestPolicy())
+                                    .build());
 
+            try {
+                submitter.submit(instance.reindexThreadRunnable);
+                // Set state to RUNNING only after successful submission
+                instance.state.compareAndSet(ThreadState.STOPPED, ThreadState.RUNNING);
```

Comment:
Will this ever be `ThreadState.PAUSED` and need to be set to running?  Seems like we should just set to running no matter what.

Reply:
The concurrency should not allow changing the state while in this method and we hare already checked for the paused state.   It should be fine how it is but the only reason to do the compareAndSet would be to check the output of the method to confirm that the state did in fact remain in the expected state and there was not some other code that changed it while processing here.    Without any check we probably are better to just set the state as a paused change would not get picked up here and the state would remain as paused even though the thread had been created.   I will fix this.